ewatercycle/models/pcrglobwb.py

```diff
@@ -11,98 +11,154 @@
 
 
+@dataclass
+class PCRGlobWBParameterSet:
+    """Parameter set for the PCRGlobWB model class.
+
+    A valid pcrglobwb parameter set consists of a folder with input data files
+    and should always include a default configuration file.
+    """
+    input_dir: Union[str, PathLike]
+    """Input folder path."""
+    default_config: Union[str, PathLike]
+    """Path to (default) model configuration file consistent with `input_data`."""
+    def __setattr__(self, name: str, value: Union[str, PathLike]):
+        self.__dict__[name] = Path(value).expanduser().resolve()
+
+
 class PCRGlobWB(AbstractModel):
     """eWaterCycle implementation of PCRGlobWB hydrological model.
 
-    Attributes
-        bmi (Bmi): GRPC4BMI Basic Modeling Interface object
-    """
-    def setup(  # type: ignore
-            self,
-            input_dir: PathLike,
-            cfg_file: PathLike,
-            additional_input_dirs: Iterable[PathLike] = [],
-            **kwargs) -> Tuple[PathLike, PathLike]:
-        """Start model inside container and return config file and work dir.
+    Args:
 
-        Args:
+        version: pick a version from :py:attr:`~available_versions`
+        parameter_set: instance of :py:class:`~PCRGlobWBParameterSet`.
+        forcing: ewatercycle forcing container;
+            see :py:mod:`ewatercycle.forcing`.
+
+    Attributes:
 
-            - input_dir: main input directory. Relative paths in the cfg_file
-            should start from this directory.
+        bmi (Bmi): GRPC4BMI Basic Modeling Interface object
+    """
+    available_versions = ('setters', )
 
-            - cfg_file: path to a valid pcrglobwb configuration file,
-            typically somethig like `setup.ini`.
+    def __init__(self, version: str, parameter_set: PCRGlobWBParameterSet,
+                 forcing: PCRGlobWBForcing):
+        super().__init__()
 
-            - additional_input_dirs: one or more additional data directories
-            that the model will have access to.
+        self.version = version
+        self.parameter_set = parameter_set
+        self._additional_input_dirs: Iterable[PathLike] = []
 
-            - **kwargs (optional, dict): any settings in the cfg_file that you
-            want to overwrite programmatically. Should be passed as a dict,
-            e.g. `meteoOptions = {"temperatureNC": "era5_tas_1990_2000.nc"}`
-            where meteoOptions is the section in which the temperatureNC option
-            may be found.
+        self._set_docker_image()
 
-        Returns: Path to config file and work dir
-        """
         self._setup_work_dir()
-        self._setup_config(cfg_file, input_dir, **kwargs)
-        self._start_container(input_dir, additional_input_dirs)
+        self._setup_default_config()
+
+        self._update_config(
+            meteoOptions={
+                "temperatureNC": forcing.temperatureNC,
+                "precipitationNC": forcing.precipitationNC,
```

Comment:
My other inline suggestion is probably caused here. We should append `forcing.directory`.

Something like
```suggestion
                "temperatureNC": str(Path(forcing.directory) / forcing.temperatureNC),
                "precipitationNC": str(Path(forcing.directory) / forcing.precipitationNC),
```

Reply:
Right! Thanks for the elaborate feedback :-)